build-tests/heft-typescript-v4-test/package.json

```diff
@@ -12,7 +12,8 @@
   },
   "devDependencies": {
     "@microsoft/api-extractor": "workspace:*",
-    "local-eslint-config": "workspace:*",
```

Comment:
Heft loads the `@typescript-eslint/parser` from `local-eslint-config`, which resolves `typescript` from that location. This causes ESLint to have a different version of TypeScript from Heft, which causes the parser to crash (because the AST data structures are different). It is probably only noticeable in major TypeScript upgrades.